File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    // Brute Force
    let Sec = 0;
    for (let i = 0; i < nums.length - 1; i++) {
        Sec = target - nums[i]
        for (let j = i + 1; j < nums.length; j++) {
            if (nums[j] == Sec) {
                return [i, j]
            }
        }
    }
};

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> fs;

        for(int i=0;i<nums.size();i++){
            int find = target - nums[i];
            if(fs.find(find) == fs.end()){
                fs[nums[i]] = i;
            }
            else{
                return {fs[find],i};
            }
        }
        return {-1,-1};
    }
};